File: java/0014-longest-common-prefix/Solution.java
import java.util.Arrays;

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String first = strs[0];
        String last = strs[strs.length - 1];

        int pos = 0;
        while (pos < first.length()) {
            if (first.charAt(pos) != last.charAt(pos)) {
                break;
            }

            ++pos;
        }

        return first.substring(0, pos);
    }
}

File: c++/0014-longest-common-prefix/solution.cpp
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string first = strs[0];
        string last = strs[strs.size() - 1];

        int pos = 0;
        while (pos < first.size()) {
            if (first[pos] != last[pos]) {
                break;
            }

            ++pos;
        }

        return first.substr(0, pos);
    }
};
